app-api-cli/src/main.rs

```diff
@@ -18,21 +20,31 @@ pub struct Cli {
 
 impl Cli {
     pub async fn run(&self) -> Result<()> {
+        let mut chunker = SimpleChunker::new(60);
         let command = Message::ApplicationAPI(self.command.clone());
         let cmd_str = serde_json::to_string(&command)?;
         println!("Transmitting: {}", &cmd_str);
         let target_address: SocketAddr = self.instance_addr.parse()?;
         let bind_address: SocketAddr = "127.0.0.1:0".parse()?;
         let socket = UdpSocket::bind(&bind_address).await?;
-        socket.send_to(&command.to_bytes(), target_address).await?;
+        for chunks in chunker.chunk(command).unwrap() {
+            socket.send_to(&chunks, target_address).await?;
+        }
+
         if self.listen {
-            let mut buf = vec![0; 1024];
-            if let Ok(len) = socket.recv(&mut buf).await {
-                let mut databuf = &buf[..len];
-                match Message::decode(&mut databuf) {
-                    Ok(msg) => println!("{msg:?}"),
-                    Err(e) => println!("{e:?}"),
+            loop {
+                let mut buf = vec![0; 1024];
+                if socket.recv(&mut buf).await.is_ok() {
+                    match chunker.unchunk(&buf) {
+                        Ok(Some(msg)) => {
+                            println!("{msg:?}");
+                            return Ok(());
+                        }
+                        Ok(None) => {}
```

Comment:
Does this mean we received a message but there is no message?

Reply:
Almost but not quite. This means we haven't been able to assemble a message yet, and also we haven't encountered any assembly or deserialization errors. It's the valid state of "I've received a few chunks which look good, but they don't add up to a full message yet".